ods_ci/tasks/Resources/RHODS_OLM/install/oc_install.robot

```diff
@@ -260,11 +261,17 @@ Apply DataScienceCluster CustomResource
         ${yml} =    Get File    ${file_path}dsc_apply.yml
         Log To Console    Applying DSC yaml
         Log To Console    ${yml}
-        ${return_code}    ${output} =    Run And Return Rc And Output    oc apply -f ${file_path}dsc_apply.yml
-        Log To Console    ${output}
-        Should Be Equal As Integers  ${return_code}  0  msg=Error detected while applying DSC CR
-        #Remove File    ${file_path}dsc_apply.yml
-        Wait For DSC Conditions Reconciled    ${OPERATOR_NS}     ${DSC_NAME}
+        ${rc}=    Set Variable    1
+        TRY
+            WHILE    ${rc} != 0    limit=2m
+                Sleep    5s
+                ${rc}    ${output}=    Run And Return Rc And Output
+                ...    oc apply -f ${file_path}dsc_apply.yml
+            END
+        EXCEPT    WHILE loop was aborted    type=start
+            Fail    msg=Error detected while applying DSC CR
+        END
```

Comment:
cannot you use Wait Until Keyword Succeedes?

Reply:
Also if your problem is that the CRD is not present, you can first check its existence and then apply the yaml

Reply:
This is to make the test case less prone to failures, not only about the CRD but I noticed that sometimes it failed at first try and then by manually doing the oc apply -f it worked just some seconds after, so this is to give a few more tries